### ISSUE TITLE
Double false positive (S2583 and S1854)

### ISSUE THREAD

**Issue by markusschaber** [SonarSource/sonar-dotnet#3386]
### Description

The code below is a minimal repro of a much more complex examle, which triggers false positive warnings for S2583 and S1854.

### Repro steps
```
namespace StaticAnalyzerProblems
{
    using System;
    using System.Diagnostics;
    using System.Threading;

    public class LoopProblem
    {
        public static void LoopMethod()
        {
            var skipSleep = true;
            while (true)
            {
                // No sleep for initial connect, but sleep for retries
                if (skipSleep)  // <- S2583 
                    skipSleep = false; // <- S1854
                else
                    Thread.Sleep(42);

                try
                {
                    MayThrow();
                    skipSleep = true; // Reset sleep only if no exception happened!
                }
                catch (Exception ex)
                {
                    Trace.WriteLine(ex);
                }
            }
        }

        public static void MayThrow()
        {
            if (DateTime.UtcNow.DayOfWeek == DayOfWeek.Sunday)
                throw new InvalidOperationException("No work on sundays!");
        }
    }
}
```

### Expected behavior
No warning in that code, both are false positives

### Actual behavior

It warns on `if (skipSleep)` and `skipSleep = false;`

### Known workarounds

I don't know any workaround.

### Related information

* C#/VB.NET Plugins Version
SonarLint for Visual Studio   4.22.0.18519

* Visual Studio Version
Microsoft Visual Studio Professional 2019
Version 16.6.2

* MSBuild / dotnet version
  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <LangVersion>8.0</LangVersion>
  </PropertyGroup>

* If running through the Scanner for MSBuild, its version
* Operating System
Windows 10 Pro, Version 1809, Build 17763.1282

**Comment by pavel-mikula-sonarsource** [comment]
Hi @markusschaber ,

Thank you for reporting this issue and narrowing it to the minimal reproducer. The root cause for both problems is same as #2600 so I'll close this report and put a note there.

**Comment by markusschaber** [comment]
It's a bit strange to leave the issue open while it's not fully fixed (in our company, we link them as "duplicate" or "depends", and only close them when the underlying issue is fixed). :-)

But thanks for confirming, I'll monitor #2600
